src/AudioResampler.cpp

```diff
@@ -74,9 +74,9 @@ AudioResampler::~AudioResampler()
 void AudioResampler::SetBuffer(AudioSampleBuffer *new_buffer, double sample_rate, double new_sample_rate)
 {
 	if (sample_rate <= 0)
-		sample_rate == 44100;
+		sample_rate = 44100;
 	if (new_sample_rate <= 0)
-		new_sample_rate == 44100;
+		new_sample_rate = 44100;
```

Comment:
I can't believe `g++` doesn't catch this and throw up _some_ sort of warning... makes me wonder if we're missing some compiler flags, and don't have all of the sanity-checking warnings enabled that we should?

Reply:
Hah! Turns out `-Wall` totally _would_ have caught that... if we were building with `-Wall`. Which we're not.

```console
$ cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_CXX_FLAGS=-Wall ..
$ # [...]
$ make VERBOSE=1
$ # [...]
[ 10%] Building CXX object src/CMakeFiles/openshot.dir/AudioResampler.cpp.o
cd /home/ferd/rpmbuild/REPOS/libopenshot/build/src && /usr/lib64/ccache/c++  -DMAGICKCORE_HDRI_ENABLE=0 -DMAGICKCORE_QUANTUM_DEPTH=16 -DOPENSHOT_IMAGEMAGICK_COMPATIBILITY=0 -DQT_CORE_LIB -DQT_GUI_LIB -DQT_MULTIMEDIAWIDGETS_LIB -DQT_MULTIMEDIA_LIB -DQT_NETWORK_LIB -DQT_WIDGETS_LIB -DUSE_IMAGEMAGICK=1 -Dopenshot_EXPORTS -I/usr/include/ImageMagick-6 -I/usr/include/ffmpeg -I/usr/include/libopenshot-audio -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtCore -isystem /usr/lib64/qt5/mkspecs/linux-g++ -isystem /usr/include/qt5/QtMultimedia -isystem /usr/include/qt5/QtNetwork -isystem /usr/include/qt5/QtMultimediaWidgets  -Wall           -g -ggdb  -fopenmp  -g -fPIC   -fPIC -std=c++11 -o CMakeFiles/openshot.dir/AudioResampler.cpp.o -c /home/ferd/rpmbuild/REPOS/libopenshot/src/AudioResampler.cpp
/home/ferd/rpmbuild/REPOS/libopenshot/src/AudioResampler.cpp: In member function ‘void openshot::AudioResampler::SetBuffer(juce::AudioSampleBuffer*, double, double)’:
/home/ferd/rpmbuild/REPOS/libopenshot/src/AudioResampler.cpp:77:15: warning: statement has no effect [-Wunused-value]
   77 |   sample_rate == 44100;
      |   ~~~~~~~~~~~~^~~~~~~~
/home/ferd/rpmbuild/REPOS/libopenshot/src/AudioResampler.cpp:79:19: warning: statement has no effect [-Wunused-value]
   79 |   new_sample_rate == 44100;
      |   ~~~~~~~~~~~~~~~~^~~~~~~~
```

...Why aren't we building with `-Wall` (at the very _least_)?

Reply:
Sounds like we need Travis and GitLab to be building with `-DCMAKE_CXX_FLAGS=-Wall`